aptos-move/framework/aptos-token/sources/token.move

```diff
@@ -586,6 +604,36 @@ module aptos_token::token {
         direct_deposit(to, token);
     }
 
+
+    /// Token owner can create this one-time withdraw capability with an expiration time
+    public fun create_withdraw_capability(
```

Comment:
Why not extract the token and keep it in the auction's account? Auction house can be hosted in a new resource account.

Reply:
no, we want the owner to still have the token until the auction completes. So they can still use/show their token before actual sale happens

Reply:
But what if they then transfer it somewhere else? Wouldn't the auction not be honored?

Reply:
The auction will fail. That is the auction house's responsibility to verify the listing

Reply:
They can't verify the listing if the token owner transfers close to the end of the auction. This seems a bit brittle in my opinion. What's wrong with the token owner giving up temporary custody during the auction?

Reply:
sorry, I think this is quite common in NFT marketplace. people don't put their NFT to the auction house escrow. There are many way for the auction house to ensure the high quality of listing. They can just penalize the user with bad behavior by blacklist them or create a reputation score, etc....

Reply:
btw: this discussion is irrelevant to the this PR for supporting withdraw with a capability right?

Reply:
@movekevin , while I agree with you, these are product aspects. I'm fine with exploring both paths, but I ultimately really dislike the idea of an auction that doesn't contain the item. It seems to violate my notion of web3 :P

Reply:
I still think it's strange that the auction can happen and buyers commit money into it, and then the seller just refuses to sell at the end by transferring the token away.